gadget/install/partition.go

```diff
@@ -56,11 +56,20 @@ func reloadPartitionTable(gadgetRoot string, device string) error {
 	}
 }
 
-// createMissingPartitions creates the partitions listed in the laid out volume
+type Constraints struct {
+	// Allow creating all partitions not just role-* ones
+	AllPartitions bool
+}
```

Comment:
This looks more like an "Options" structure than a "Constraints", but if the plan is to use it to host constraints in the future, then no complaints :-)